### DISCUSSION TITLE
StowRS - 202 accepted

### DISCUSSION THREAD

**Discussion by jcomedouteau** [dcm4che/dcm4chee-arc-light#4021] [Q&A] [answered]
Hi,
We are currently using dcm4chee with docker, embedded in a webapp based on java.
As stowRS was not available when we made our implementation, we decided to use storescu to upload dicoms into the pacs.
We are now encountering the problem described in this issue:
https://github.com/dcm4che/dcm4che/issues/666
And we want to be able to tell when the data storage went bad.

So we decided to move to the REST implementation, and we now have another problem.
We receive as an answer "202 - ACCEPTED" responses with payloads as described further.
I checked the code and it seems that there was a failure during storage ? How can we decypher this response ? Is it even relevant? (as it seems that, when downloading from the PACS, all the data is present). 

Thanks for your answers.

The response:
```{
   "00081190":{
      "vr":"UR",
      "Value":[
         "http://XXXX/dcm4chee-arc/aets/XXXX/rs/studies/1.4.9.12.34.1.8527.2179683266094210009464095289809241350809"
      ]
   },
   "00081198":{
      "vr":"SQ",
      "Value":[
         {
            "00081150":{
               "vr":"UI",
               "Value":[
                  "1.2.40.0.13.1.15.110.3.165.2"
               ]
            },
            "00081155":{
               "vr":"UI",
               "Value":[
                  "2.25.202784732260572757079300619503889468793"
               ]
            },
            "00081197":{
               "vr":"US",
               "Value":[
                  50329
               ]
            }
         },
         {
            "00081150":{
               "vr":"UI",
               "Value":[
                  "1.2.40.0.13.1.15.110.3.165.2"
               ]
            },
            "00081155":{
               "vr":"UI",
               "Value":[
                  "2.25.310922709863880872173886417845302976884"
               ]
            },
            "00081197":{
               "vr":"US",
               "Value":[
                  50329
               ]
            }
         }
      ]
   },
   "00081199":{
      "vr":"SQ",
      "Value":[
         {
            "00081150":{
               "vr":"UI",
               "Value":[
                  "1.2.840.10008.5.1.4.1.1.4"
               ]
            },
            "00081155":{
               "vr":"UI",
               "Value":[
                  "1.4.9.12.34.1.8527.6037536041694311122163754668844248392857"
               ]
            },
            "00081190":{
               "vr":"UR",
               "Value":[
                  "http://xxx/dcm4chee-arc/aets/XXX/rs/studies/1.4.9.12.34.1.8527.2179683266094210009464095289809241350809/series/1.4.9.12.34.1.8527.9295863121584039014363895718080173225361/instances/1.4.9.12.34.1.8527.6037536041694311122163754668844248392857"
               ]
            }
         }
      ]
   }
}
```

**Comment by vrindanayak** [comment, marked as answer]
`202 - Accepted` response is sent back when some objects were stored successfully (indicated in `Referenced SOP Sequence` - `00081199`) and some failed (indicated in `Failed SOP Sequence` - `00081198`).

`50329` in `Failure Reason` - `00081197` indicates archive failed to parse DICOM stream. - See https://github.com/dcm4che/dcm4chee-arc-light/issues/3517

Refer [Storage Application C-STORE Response Status Return Reasons](https://dcm4chee-arc-cs.readthedocs.io/en/latest/networking/specs/storage/storage.html#id8)
